tests/contrib/operators/test_kubernetes_job_operator.py

```diff
@@ -0,0 +1,116 @@
+            self.assertEquals('success', task_instance.current_state())
+        else:
+            # Failed pod will cause task_instance to fail
+            with self.assertRaises(Exception):
+                task_instance.run()
+            self.assertEquals('failed', task_instance.current_state())
+
+    def test_basic_execution_failure(self):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_dictionary={
+                                                            'containers': [self.CONTAINER]
+                                                         },
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=False,
+                                               kubernetes_job_operator_task=task)
+
+    @mock.patch('subprocess.check_output')
+    def test_basic_execution_success(self, mock_check_output):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_dictionary={
+                                                            'containers': [self.CONTAINER]
+                                                         },
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=True,
+                                               kubernetes_job_operator_task=task)
+
+    @mock.patch('subprocess.check_output')
+    def test_user_provided_yaml_execution_failure(self, mock_check_output):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_template=self.YAML,
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=False,
+                                               kubernetes_job_operator_task=task)
+
+    @mock.patch('subprocess.check_output')
+    def test_user_provided_yaml_execution_success(self, mock_check_output):
+        task = kubernetes_operator.KubernetesJobOperator(kubernetes_job_name='test-basic',
+                                                         kubernetes_job_yaml_template=self.YAML,
+                                                         task_id=self.TASK_ID,
+                                                         dag=self.DAG)
+        self.kubernetes_job_operator_execution(success=True,
+                                               kubernetes_job_operator_task=task)
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
nice! ALso that reminds me: `python -m unittest.discover` might be good enough for this repo for now! Its built it and should mostly work. I might have the exact syntax wrong though